### PR TITLE
Play 2.4.0

### PR DESCRIPTION
**Pull request by Andrei-Pozolotin**
May be it's time to take a look on play 2.4? 
I would hope securesocial 3.0 to support both paly 2.3 and 2.4.
This pull merely shows changes needed to make securesocial master to build with play 2.4 - not much.

### CODE REVIEW

**Comment by sheldonkreger** [comment]
Any further progress on 2.4 support?

**Comment by hohl** [comment]
Play 2.4 is released. Maybe it's time to merge this one?

**Comment by hohl** [comment]
Since Play 2.4 is released. Any progress on this?

**Comment by nafg** [comment]
+1

**Comment by Andrei-Pozolotin** [comment]
the time is now :-)
https://groups.google.com/forum/#!topic/play-framework-dev/De4-4sp2fL0

**Comment by keepscoding** [comment]
+1

**Comment by hohl** [comment]
@Andrei-Pozolotin Would this pull-request still work if the merge conflicts are resolved? Tried to add support by myself but failed as you may have read in the referenced issue.

**Comment by Andrei-Pozolotin** [comment]
@hohl do not know; I did it once, some one needs to do it second time :-)

File: module-code/test/scenarios/helpers/package.scala [hunk 1]
```diff
@@ -20,14 +20,15 @@ package object helpers {
...
+    // override 
+    def getControllerInstance[A](controllerClass: Class[A]): A = {
       val instance  = controllerClass.getConstructors.find { c =>
         val params = c.getParameterTypes
         params.length == 1 && params(0) == classOf[RuntimeEnvironment[U]]
       }.map {
         _.asInstanceOf[Constructor[A]].newInstance(runtimeEnvironment)
       }
-      instance.getOrElse(super.getControllerInstance(controllerClass))
+      ??? //instance.getOrElse(super.getControllerInstance(controllerClass))
```
Comment (lloydmeta) [inline module-code/test/scenarios/helpers/package.scala]: Just wanted to highlight this; not sure if this was forgotten..

**Comment by elvenking** [comment]
+1

**Comment by jmswenski** [comment]
+1

**Comment by mrkaspa** [comment]
Any progress I need this for my play 2.4 application

**Comment by agapep** [comment]
+1

**Comment by eximius313** [comment]
+1

**Comment by jaliss** [comment]
master-SNAPSHOT is 2.4 compatible now.

**Comment by MipH** [comment]
I've used master-SNAPSHOT in my project for Play 2.3
No it is broken (because of 2.4 merge). Should I use 3.0-M3 instead?

**Comment by jaliss** [comment]
@MipH yes, before adding 2.4 compatibility what was there in master was exactly the same code as in 3.0-M3.  Migrate to that and you should be ok.

**Pull request closed without merging**